db/memtable.cc

```diff
@@ -929,6 +929,18 @@ void MemTable::MultiGet(const ReadOptions& read_options, MultiGetRange* range,
 
     if (found_final_value) {
       iter->value->PinSelf();
+      if (read_options.value_size > 0) {
+        if (range->GetValueSize() + iter->value->size() >
+            read_options.value_size) {
+          while (iter != temp_range.end()) {
```

Comment:
```range``` may not cover all keys in the MultiGet batch, just the ones that could be present in memtable. It would be better to move the logic to abort the request and set the status of remaining keys in ```MultiGetContext```, so it can set status for all remaining keys.

Reply:
I'm using MultiGetContext::value_size to hold the value_size of keys processed it now.  VersionSet::MultiGet then checks the MultiGetContext::value_size  (updated by MemTable::MultiGet) and then sets the remaining keys status to Abort if size exceeds. So this way memtable keys and sst file keys both are processed.

But it makes sense to not even call VersionSet::MultiGet if size exceeds and MemTable::MultiGet should update the remaining keys. So i will change the logic to do so. Thanks.